### PR TITLE
[Hotfix] Fix empty row bug in balances

### PR DESCRIPTION
**Pull request by W3stside**
## adds `filterAndFillTokenDetails` fn to fill tokens that do not have pre-filled data when pulled from localStorage

Why? fixes empty row bug in balances

Problem was:
Screenshot from 2020-11-30 16-22-01

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr1646--dexreact.review.gnosisdev.com)**: Testing web app
   * **📚 [Storybook](https://pr1646--dexreact.review.gnosisdev.com/storybook/)**: Component stories

File: src/services/factories/tokenList.ts [hunk 1]
```diff
@@ -284,6 +288,32 @@ export function getTokensFactory(factoryParams: {
...
+    const tokensWithNoDetails: TokenDetails[] = []
+    const tokensWithDetails = tokensConfig.reduce<TokenDetails[]>((acc, token) => {
+      if (!token.name || !token.symbol) {
+        tokensWithNoDetails.push(token)
+        return acc
+      }
+
+      acc.push(token)
+      return acc
+    }, [])
```
Comment (Velenir) [inline src/services/factories/tokenList.ts:301]: Not sure there's much advantage to `reduce` over a for loop if you do side effects (`tokensWithNoDetails.push(token)`) anyways

File: src/services/factories/tokenList.ts [hunk 2]
```diff
@@ -284,6 +288,32 @@ export function getTokensFactory(factoryParams: {
...
+
+    let updatedTokens: TokenDetails[] = []
+
+    if (tokensWithNoDetails.length > 0) {
```
Comment (Velenir) [inline src/services/factories/tokenList.ts:305]: would 
```ts
if (tokensWithNoDetails.length === 0) return tokensWithDetails
```
read simpler?

File: src/services/factories/tokenList.ts [hunk 3]
```diff
@@ -266,8 +266,12 @@ export function getTokensFactory(factoryParams: {
...
   }
 
-  async function updateTokenDetails(networkId: number, numTokens: number): Promise<void> {
-    const tokensConfig = tokenListApi.getTokens(networkId)
+  async function updateTokenDetails(
+    networkId: number,
+    numTokens: number,
+    tokensConfig?: TokenDetails[],
+  ): Promise<void> {
+    tokensConfig = (tokensConfig?.length && tokensConfig) || tokenListApi.getTokens(networkId)
```
Comment (Velenir) [inline src/services/factories/tokenList.ts:274]:
```suggestion
   if (tokensConfig.length === 0) tokensConfig = tokenListApi.getTokens(networkId)
```
maybe

**Review by Velenir — APPROVED** [review]
Great! Works for me :+1:

File: src/services/factories/tokenList.ts (hunk 3 above)
Comment (W3stside) [reply src/services/factories/tokenList.ts:274]: with default assignment = [] in args yeah good idea

File: src/services/factories/tokenList.ts (hunk 2 above)
Comment (W3stside) [reply src/services/factories/tokenList.ts:305]: yep!

**Pull request merged**